android-project/app/src/main/java/org/diasurgical/devilutionx/DevilutionXSDLActivity.java

```diff
@@ -79,11 +79,12 @@ private boolean isMissingGameData() {
 		if (lang.startsWith("ru") && !fileManager.hasFile("ru.mpq"))
 			return true;
 		if (lang.startsWith("ko") || lang.startsWith("zh") || lang.startsWith("ja")) {
-			if (!fileManager.hasFile("fonts.mpq"))
+			if (!fileManager.hasFile("fonts.mpq") ||
+					fileManager.fileSize("fonts.mpq") == 70471463 /* v1 */ ||
+					fileManager.fileSize("fonts.mpq") == 53991069 /* v2 */ ||
+					fileManager.fileSize("fonts.mpq") == 58488019 /* v3 */) {

```

Comment:
`fonts.mpq` version was deliberately checked regardless of language settings because an out-of-date file needs to be updated regardless.